tools/retrive_go_repos.py

```diff
@@ -0,0 +1,302 @@
+"""Get list of repo urls for a given organization with 1st level dependancy for GO language."""
+
+import json
+import logging
+from datetime import datetime, timedelta
+from urllib.request import urlopen, URLError
+
+import requests
+from bs4 import BeautifulSoup
+from dateutil.parser import isoparse
+import time
+
+"""
+Configure below variables, which are used in get repo with dependancy details for GO.
+--------------------------------------------------------------------------------------------
+GITHUB_ACCESS_TOKEN: Github token to query github api
+REPO_UPDATED_WITHIN_N_DAYS: if None set then will retrive all repos
+                            else will get only repos those updated in given N days
+ORGANIZATION: organition name (Ex openshift/knative/kubevirt)
+
+Note: We are filtering out disabled and archived repos.
+"""
+GITHUB_ACCESS_TOKEN = ""
+REPO_UPDATED_WITHIN_N_DAYS = None
+ORGANIZATION = "kubevirt"
+LANGUAGE = "GO"
+ARCHIEVED = False
+DISABLED = False
+
+# set logging level to INFO
+logging.getLogger().setLevel(logging.INFO)
+
+
+pkg_list = []
+repo_urls = []
+dependancy_urls = []
+
+
+def check_repo_updated_date(date: str) -> bool:
```

Comment:
Can you explain what the logic here is a bit?

Reply:
If REPO_UPDATED_WITHIN_N_DAYS is not None then we are checking passed date is within specified days in REPO_UPDATED_WITHIN_N_DAYS, if None then no need to check condition.